src/io.c

```diff
@@ -0,0 +1,2024 @@
+            stream->hints[i] = stream->hints[i + 3];
+        nr--;
+    }
+    /* if pos is same as before only relevant if new entry has higher
+       priority */
+    if (nr > 0 && stream->hints[3 * (nr - 1)] == pos)
+        nr--;
+
+    if (stream->indent < pos &&
+        (stream->hints[3 * nr] == -1 || val < stream->hints[3 * (nr) + 1])) {
+        stream->hints[3 * nr] = pos;
+        stream->hints[3 * nr + 1] = val;
+        stream->hints[3 * nr + 2] = stream->indent;
+        stream->hints[3 * nr + 3] = -1;
+    }
+    stream->indent += indentdiff;
+}
+/* helper function to find line break position,
+   returns position nr in stream[hints] or -1 if none found */
+Int nrLineBreak(KOutputStream stream)
+{
+    Int nr = -1, min, i;
+    for (i = 0, min = INT_MAX; stream->hints[3 * i] != -1; i++) {
+        if (stream->hints[3 * i] > 0 &&
+            stream->hints[3 * i + 1] - stream->hints[3 * i] <= min) {
+            nr = i;
+            min = stream->hints[3 * i + 1] - stream->hints[3 * i];
+        }
+    }
+    if (min < INT_MAX)
+        return nr;
+    else
+        return -1;
+}
+
+
+void PutChrTo(KOutputStream stream, Char ch)
+{
+    Int  i, hint, spos;
+    Char str[MAXLENOUTPUTLINE];
+
+
+    /* '\01', increment indentation level                                  */
+    if (ch == '\01') {
+
+        if (!stream->format)
+            return;
+
+        /* add hint to break line  */
+        addLineBreakHint(stream, stream->pos, 16 * stream->indent, 1);
+    }
+
+    /* '\02', decrement indentation level                                  */
+    else if (ch == '\02') {
+
+        if (!stream->format)
+            return;
+
+        /* if this is a better place to split the line remember it         */
+        addLineBreakHint(stream, stream->pos, 16 * stream->indent, -1);
+    }
+
+    /* '\03', print line                                                   */
+    else if (ch == '\03') {
+
+        /* print the line                                                  */
+        if (stream->pos != 0) {
+            stream->line[stream->pos] = '\0';
+            PutLineTo(stream, stream->pos);
+
+            /* start the next line                                         */
+            stream->pos = 0;
+        }
+        /* reset line break hints                                          */
+        stream->hints[0] = -1;
+    }
+
+    /* <newline> or <return>, print line, indent next                      */
+    else if (ch == '\n' || ch == '\r') {
+
+        /* put the character on the line and terminate it                  */
+        stream->line[stream->pos++] = ch;
+        stream->line[stream->pos] = '\0';
+
+        /* print the line                                                  */
+        PutLineTo(stream, stream->pos);
+
+        /* and dump it from the buffer */
+        stream->pos = 0;
+        if (stream->format) {
+            /* indent for next line                                         */
```

Comment:
Note the uneven alignment of the `*/` compared to the previous lines (which of course is not your fault). I don't know why `clang-format` isn't removing those -- but if we run `clang-format` anyway, I'd consider running sed with `s;\s+\*/; */;g` or so, to simply remove this annoying-to-maintain manual alignment.